app/src/main/java/org/cru/godtools/activity/BasePlatformActivity.java

```diff
@@ -242,16 +247,65 @@ private void setupNavigationDrawer() {
         }
     }
 
+    /**
+     * This method is used to update the Navigation Draws Login settings.
+     * It will only show log in information if the device is set to english
+     * and display log in links based on users status.
+     * Updated by:  Gyasi Story
+     */
     private void updateNavigationDrawerMenu() {
-        if (mLoginItem != null) {
-            mLoginItem.setVisible(mTheKey.getDefaultSessionGuid() == null);
-        }
-        if (mSignupItem != null) {
-            mSignupItem.setVisible(mTheKey.getDefaultSessionGuid() == null);
+        if (!containsLoginLanguage()) {  // For non English
+            if (mLoginItem != null) {
+                mLoginItem.setVisible(false);
+            }
+            if (mSignupItem != null) {
+                mSignupItem.setVisible(false);
+            }
+            if (mLogoutItem != null) {
+                mLogoutItem.setVisible(false);
+            }
+
+        } else { // For English
+            Boolean isNotLoggedIn = mTheKey.getDefaultSessionGuid() == null;
+            if (mLoginItem != null) {
+                mLoginItem.setVisible(isNotLoggedIn);
+            }
+            if (mSignupItem != null) {
+                mSignupItem.setVisible(isNotLoggedIn);
+            }
+            if (mLogoutItem != null) {
+                mLogoutItem.setVisible(!isNotLoggedIn);
+            }
         }
-        if (mLogoutItem != null) {
-            mLogoutItem.setVisible(mTheKey.getDefaultSessionGuid() != null);
+    }
+
+    /**
+     * This method will determine if the application contains
+     * a supported language for Login Notifications. As we
+     * get more support for more languages you can add them to {@code languages}.
+     *
+     * @return Boolean value if login Language
+     *
+     * Updated by: Gyasi Story
+     */
+    private boolean containsLoginLanguage() {
+        // List of supported Login Languages
+        String[] languages = {
+                Locale.ENGLISH.getLanguage()
+        };
+
+        LocaleListCompat localeListCompat = ConfigurationCompat
+                .getLocales(Resources.getSystem().getConfiguration());
+        for (int i = 0; i < localeListCompat.size(); i++) {
+            for (String lang : languages) {
+                if (localeListCompat.get(i).getLanguage().equals(lang)) {
+                    return true;
+                }
+            }
         }
+
+        return false;
+
```

Comment:
would this work instead?
```
LocaleListCompat localeListCompat = ConfigurationCompat
                .getLocales(Resources.getSystem().getConfiguration());
return localeListCompat.getFirstMatch(languages) != null;
```

Reply:
@frett That didn't work because it default to English instead of returning a null value.

Reply:
k, the stream idea I had won't work either since there isn't a direct way to get a raw list of languages out of LocaleListCompat. so, this should be good as is